relayer/relays/beacon/main.go

```diff
@@ -110,15 +110,34 @@ func (r *Relay) Sync(ctx context.Context) error {
 		}
 	}
 
-	logrus.Info("done with sync committee updates ")
+	lastFinalizedHeader, err := r.writer.getLastStoredFinalizedHeader()
+	if err != nil {
+		logrus.WithError(err).Error("unable to get last finalized header")
 
-	logrus.Info("starting to sync finalized headers")
+		return err
+	}
 
-	_, _, err = r.SyncFinalizedHeader(ctx)
+	lastFinalizedSlot, err := r.writer.getLastStoredFinalizedHeaderSlot()
 	if err != nil {
+		logrus.WithError(err).Error("unable to get last finalized header slot")
+
 		return err
 	}
 
+	r.syncer.Cache.FinalizedHeaders = append(r.syncer.Cache.FinalizedHeaders, lastFinalizedHeader)
+
+	logrus.WithFields(logrus.Fields{
+		"hash": lastFinalizedHeader,
+		"slot": lastFinalizedSlot,
+	}).Info("set cache: last finalized header")
+
+	logrus.Info("starting to sync finalized headers")
+
+	err = r.SyncHeaders(ctx)
+	if err != nil {
+		logrus.WithError(err).Error("error while syncing headers")
+	}
```

Comment:
need to return the error if its not nil.

Reply:
Addressed in https://github.com/Snowfork/snowbridge/pull/668/commits/27119fdd6815bfb60e1684778a3347fa3d6a1582.